src/main/resources/api.yaml

```diff
@@ -89,6 +87,30 @@ paths:
           description: User is not authorized
         404:
           description: Project not found
+  /sprints:
+    get:
+      tags:
+        - sprints
+      summary: Obtains sprint ids by their state
+      operationId: getSprints
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+        - in: query
+          name: sprintState
+          required: false
+          type: array
+          items:
+            $ref: '#/definitions/SprintStateDto'
+      responses:
+        200:
+          description: List of sprint ids with requested state
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/SprintIdWithStateDto'
```

Comment:
Dlaczego nie zwracamy pełnego sprintu?

Reply:
To generalnie służy tylko do tego, że potem sobie można robić GET na `/projects/{id}/sprints/{id}/availability` i na `/projects/{id}/sprints/{id}/summary` - cały sprint w sumie nie jest do niczego potrzebny, ale no można by zwracać cały

Reply:
Może raczej bardziej mi chodziło o nazwę, jak będziemy potrzebować np. description to zmienimy na `SprintIdWithStateAndDescriptionDto`? ;)

Reply:
Buildery takie fajne są podobno

Reply:
Znaczy na pewno powstanie pełny sprint (z datami i tak dalej) do tworzenia go

Reply:
No i będą jakieś takie różne podsumowania (z przeliczonymi współczynnikami i innymi takimi) zależnie czy `summary`, czy `availability`

Reply:
Mogę stworzyć pełne `SprintDto` w sumie, ale to będę musiała zrozumieć kwestię dat najpierw

Reply:
No to już jak uważasz

Reply:
O nie, przebrnęłam chyba przez daty, ale tam jest DURATION

Reply:
To chyba w godzinach? Chociaż był szalony pomysł na kwadranse/połówki?

Reply:
Aaa, ja wiem czemu!

Reply:
Normalny sprint przecież nie ma stanu

Reply:
Znaczy można dorzucić, ale to nie będzie to samo `SprintDto` co przy tworzeniu

Reply:
`SprintDto` i `SprintWithStateDto`

Reply:
To nie wiem

Reply:
No to albo `SprintDto` i `SprintWithStateDto`, albo `SprintDto` i  <jakaś-nazwa> odzwierciedlająca stan + id

Reply:
> Może raczej bardziej mi chodziło o nazwę, jak będziemy potrzebować np. description to zmienimy na `SprintIdWithStateAndDescriptionDto`? ;)

A to było podchwytliwe, BO SPRINT NIE MA DESCRIPTION

Reply:
To ja poczekam, aż ktoś mi powie, jak mam żyć

Reply:
można zrobić typ zwracany inline (obiekt z wymaganymi propertiesami), więc nazwa nada się sama i nikt nie będzie miał do nikogo pretensji

Reply:
Wydaje mi się, że nie powinniśmy za bardzo komplikować interfejsu i zrobić `SprintDto` ze stanem, ale przy tworzeniu sprintu wymagać żeby go nie było lub żeby był początkowy. Jak ja robiłem projekty to nie robiłem `ProjectDto` i `ProjectWithIdDto`.

Reply:
Okej, no pewnie racja, ale raczej nie chcemy tego stanu przekazywać w żadnej innej sytuacji - to jest jakaś tam dynamiczna właściwość, której nie wyjmiesz sobie z bazy ani nic. Chcemy to zawsze przeliczać, zwracając komuś `SprintDto`?

Reply:
No pewnie, zasób to zasób, robiąc request musimy zdawać sobie sprawę, że dostajemy stan zasobu na czas requestu. Tak samo musielibyśmy myśleć w przypadku każdego zasobu modyfikowalnego -- za 5 minut może być już inny. Jak dla mnie stan to część zasobu, a to że go nie trzymamy na bazie to szczegół implementacyjny, którego nie należy wystawiać na zewnątrz.

Poza tym serwer ma autonomię w ustalaniu stanu, klient nie może go sobie wyliczyć na własną rękę. Wysłanie danych nadmiarowo jest lepszym rozwiązaniem niż ich niedomiar i odpytywanie o nie osobno.

Reply:
Okej, bardziej kwestionowałam przeliczanie ciągle czegoś, co nie jest potrzebne (i dodatkowo tymczasowe), ale generalnie zgadzam się z argumentami

Trochę głupio, bo tak to było ustalane i tak figuruje w opisie issue - nie wiem, czy nie utrudni to życia autorom zależnych issue, ale może jakoś to przeżyjemy

Reply:
jeśli zamiast listy {id: stan} będzie lista całych sprintów, to chyba nawet ułatwi

Reply:
No dobra, dobra - zrobięę